src/HotChocolate/Core/src/Validation/Rules/ValueVisitor.cs

```diff
@@ -207,6 +207,10 @@ public ValueVisitor()
                     }
                 }
             }
+            else if (namedType is InputUnionType inputUnionType)
```

Comment:
The changes in this file are probably not a good idea ... we should not add so much performance implications on all the rules for a single case.